util/src/builder/callback_data.rs

```diff
@@ -0,0 +1,169 @@
+use twilight_model::{
+    application::{callback::CallbackData, component::Component},
+    channel::{
+        embed::Embed,
+        message::{AllowedMentions, MessageFlags},
+    },
+};
+
+/// Create a [`CallbackData`] with a builder.
+///
+/// # Example
+/// ```
+/// use twilight_util::builder::CallbackDataBuilder;
+/// use twilight_model::channel::message::MessageFlags;
+///
+/// let callback_data = CallbackDataBuilder::new()
+///     .content("My callback message".to_string())
+///     .flags(MessageFlags::EPHEMERAL)
+///     .build();
+/// ```
+#[derive(Debug)]
+#[allow(clippy::module_name_repetitions)]
+#[must_use = "builders have no effect if unused"]
+pub struct CallbackDataBuilder(CallbackData);
+
+impl CallbackDataBuilder {
+    /// Create a new builder to construct a [`CallbackData`].
+    pub fn new() -> Self {
+        Self::default()
+    }
+
```

Comment:
This can be const and it also clarifies the code
```suggestion
    pub const fn new() -> Self {
        Self(CallbackData {
            allowed_mentions: None,
            components: None,
            content: None,
            embeds: Vec::new(),
            flags: None,
            tts: None,
        })
    }